compactHeadersApi.js

```diff
@@ -190,6 +190,67 @@ function install(window) {
     msgHeaderView.append(compactHeadersPopup);
   }
 
+  function patchRecipientClass() {
+    window.customElements.whenDefined("header-recipient").then(classHeaderRecipient => {
+      if (!classHeaderRecipient.prototype.originalUpdateRecipient) {
+        classHeaderRecipient.prototype.originalUpdateRecipient = classHeaderRecipient.prototype.updateRecipient;
+        classHeaderRecipient.prototype.updateRecipient = function() {
+          this.originalUpdateRecipient();
+
+          if (this.dataset.headerName == "to" || this.dataset.headerName == "cc") {
+            this.nameLine.textContent = this.displayName;
+            this.addressLine.textContent = this.emailAddress;
+            if (this.displayName) {
+              this.classList.add("has-display-name");
+            } else {
+              this.classList.remove("has-display-name");
+            }
+          }
+        }
+        // Call updateRecipient for existing recipients before patched
+        for (let recipient of expandedtoBox.querySelectorAll('li[is="header-recipient"]')) {
+          recipient.updateRecipient();
+        }
+        for (let recipient of expandedccBox.querySelectorAll('li[is="header-recipient"]')) {
+          recipient.updateRecipient();
+        }
+      }
+    });
+  }
+
+  function createStyle() {
+    if (!document.getElementById("compactHeadersStyle")) {
+      let style = document.createElement('style');
+      style.id = "compactHeadersStyle";
+      style.textContent = `
+#messageHeader[compact="compact"] :is(#headerSenderToolbarContainer, #expandedtoRow, #expandedccRow) {
+  min-width: 0;
+}
+
+#messageHeader[compact="compact"] :is(#expandedtoRow, #expandedccRow) .recipients-list {
+  flex-wrap: nowrap;
+}
+
+#messageHeader[compact="compact"] :is(#expandedtoRow, #expandedccRow) .recipients-list .header-recipient {
+  min-width: max-content;
+}
+
+#messageHeader[compact="compact"].message-header-show-sender-full-address :is(#expandedtoLabel, #toHeading, #expandedccLabel, #ccHeading) {
+  align-self: center;
+}
+
+#messageHeader[compact="compact"].message-header-show-sender-full-address .has-display-name .recipient-single-line {
+  display: none;
+}
+
+#messageHeader[compact="compact"].message-header-show-sender-full-address .has-display-name .recipient-multi-line {
+  display: inline-flex;
+}
```

Comment:
If following criteria fulfilled, we show `To/CC` header as multi-line:
1. In compact mode
2. `Always show sender's full address` enabled
3. The `To`/`CC` address has display name (email-only address will still use single-line view)

Reply:
Thanks, I will test this later today.